node/src/components/storage/tests.rs

```diff
@@ -388,293 +357,346 @@ fn switch_block_for_block_header(
+                    block_header_with_metadata.block_signatures, block_signatures,
+                    "Should have retrieved expected block signatures"
+                );
+                let block_with_metadata = storage
+                    .read_block_and_sufficient_finality_signatures_by_height(
+                        block.header().height(),
+                    )
+                    .expect("should not throw exception")
+                    .expect("should not be None");
+                assert_eq!(
+                    block_with_metadata.block.header(),
+                    block.header(),
+                    "Should have retrieved expected block header"
+                );
+                assert_eq!(
+                    block_with_metadata.finality_signatures, block_signatures,
+                    "Should have retrieved expected block signatures"
+                );
+            }
+        });
     }
 }
 
 #[test]
 fn can_retrieve_block_by_height() {
-    let mut harness = ComponentHarness::default();
-    let mut storage = storage_fixture(&harness);
+    struct BlockData {
+        era_id: EraId,
+        merkle_tree_hash_activation: EraId,
+    }
 
-    // Create a random block, load and store it.
-    let block_33 = Box::new(Block::random_with_specifics(
-        &mut harness.rng,
-        EraId::new(1),
-        33,
-        ProtocolVersion::V1_0_0,
-        true,
-    ));
-    let block_14 = Box::new(Block::random_with_specifics(
-        &mut harness.rng,
-        EraId::new(1),
-        14,
-        ProtocolVersion::V1_0_0,
-        false,
-    ));
-    let block_99 = Box::new(Block::random_with_specifics(
-        &mut harness.rng,
-        EraId::new(2),
-        99,
-        ProtocolVersion::V1_0_0,
-        true,
-    ));
+    let block_specs = vec![
+        // Generates v1 blocks
+        BlockData {
+            era_id: EraId::from(1),
+            merkle_tree_hash_activation: EraId::from(5),
+        },
+        // Generates v2 blocks
+        BlockData {
+            era_id: EraId::from(1),
+            merkle_tree_hash_activation: EraId::from(0),
+        },
+    ];
 
-    // Both block at ID and highest block should return `None` initially.
-    assert!(get_block_at_height(&mut storage, 0).is_none());
-    assert!(get_block_header_at_height(&mut storage, 0).is_none());
-    assert!(get_highest_block(&mut harness, &mut storage).is_none());
-    assert!(get_block_at_height(&mut storage, 14).is_none());
-    assert!(get_block_header_at_height(&mut storage, 14).is_none());
-    assert!(get_block_at_height(&mut storage, 33).is_none());
-    assert!(get_block_header_at_height(&mut storage, 33).is_none());
-    assert!(get_block_at_height(&mut storage, 99).is_none());
-    assert!(get_block_header_at_height(&mut storage, 99).is_none());
-
-    // Inserting 33 changes this.
-    let was_new = put_block(&mut harness, &mut storage, block_33.clone());
-    assert!(was_new);
+    for block_spec in block_specs {
```

Comment:
Not sure if we really need to double all the tests.
For this one we could just set the activation at era 2, so one of the three example blocks uses the new scheme?

Reply:
Yes, that could also work.

I'll keep it as it, since it's not a blocker and is also doing the job. The test is very quick to execute so we're not adding much to the total execution of the tests.

btw: I used the approach with specifically selected eras and activation points in the tests that are long to execute (mainly in `multi_stage_test_reactor/test_chain.rs`)